src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -29,6 +29,26 @@ internal abstract class AbstractRecommendationService : IRecommendationService
             return symbols;
         }
 
+        protected static IEnumerable<ISymbol> GetRecommendedNamespaceNameSymbols(
+            SemanticModel semanticModel, SyntaxNode declarationSyntax, CancellationToken cancellationToken)
+        {
+            var containingNamespaceSymbol = semanticModel.Compilation.GetCompilationNamespace(
+                semanticModel.GetEnclosingNamespace(declarationSyntax.SpanStart, cancellationToken));
+
+            var symbols = semanticModel.LookupNamespacesAndTypes(declarationSyntax.SpanStart, containingNamespaceSymbol)
+                                       .Where(recommendationSymbol => IsNonIntersectingNamespace(recommendationSymbol, declarationSyntax));
+
+            return symbols;
+        }
+
+        protected static bool IsNonIntersectingNamespace(ISymbol recommendationSymbol, SyntaxNode declarationSyntax)
+        {
+            return recommendationSymbol.IsNamespace() &&
```

Comment:
How about this?
```
Namespace Foo.$$
    Namespace Bar
    End Namespace

    Namespace Baz
    End Namespace
End Namespace
```
Should `Baz` also show up in the list?

It's also worth noting that getting equivalent behaviour in C# will probably need additional lookups.